hopp/offshore/floating_platform.py

```diff
@@ -0,0 +1,345 @@
+                        self.depth, fab_cost, design_cost, steel_cost
+                        )
+
+        total_mass = self.mass + substructure_mass  # t
+
+         # Call the install_platform function
+        self.install_capex = install_platform(total_mass, self.area, self.distance, \
+                                                   install_duration, self.install_vessel)
+
+    # An install object needs to have attribute system_capex, installation_capex, and detailed output
+    @property
+    def system_capex(self):
+
+        return {}
+
+    @property 
+    def installation_capex(self):
+        
+        return self.install_capex
+
+    @property
+    def detailed_output(self):
+
+        return {}
+
+# Define individual calculations and functions to use outside or with ORBIT
+def calc_substructure_mass_and_cost(mass, area, depth, fab_cost=14500., design_cost=4.5e6, sub_cost=3000, pile_cost=0):
+    '''
+    Platform is substructure and topside combined
+    All funstions are based off NREL's ORBIT (oss_design)
+    default values are specified in ORBIT
+    '''
+    #Inputs needed
+    topside_mass = mass
+    topside_fab_cost_rate   =   fab_cost   
+    topside_design_cost     =   design_cost
+
+    '''Topside Cost & Mass
+    Topside Mass is the required Mass the platform will hold
+    Topside Cost is a function of topside mass, fab cost and design cost'''
+    topside_cost   =   topside_mass   *topside_fab_cost_rate  +topside_design_cost
+
+    '''Substructure
+    Substructure Mass is a function of the topside mass
+    Substructure Cost is a function of of substructure mass pile mass and cost rates for each'''
+
+    substructure_cost_rate  =   sub_cost        # USD/t
+
+    substructure_mass       =   0.4 *   topside_mass        # t
+    substructure_cost  =   (substructure_mass  *substructure_cost_rate)     # USD  
+    substructure_total_mass  =   substructure_mass    # t
+
+    '''These arrays were pulled from Rebecca Fuchs SemitTaut_mooring cost models'''
+    depths = np.array([0, 500, 750, 1000, 1250, 1500]) # [m]
+
+    anchor_costs = np.array([28823, 112766, 125511, 148703, 204988, 246655]) # [USD]
+    finterp_anchor_cost = interp1d(depths, anchor_costs)
+    anchor_cost = finterp_anchor_cost(depth)
+
+    total_line_costs = np.array([430315, 826598, 1221471, 1682208, 2380035, 3229700]) # [USD]
+    finterp_total_line_cost = interp1d(depths, total_line_costs)
+    line_cost = finterp_total_line_cost(depth)
+    
+    mooring_cost = line_cost + anchor_cost
+    
+    '''Total Platform capex = capex Topside + capex substructure'''
+    total_capex = (topside_cost + substructure_cost + mooring_cost)
+    platform_capex  = total_capex # USD
+    platform_mass   = substructure_total_mass + topside_mass    # t
+    
+    return platform_capex, platform_mass
+
+#@process
+def install_platform(mass, area, distance, install_duration=14, vessel=None):
```

Comment:
are the mass and area here just the platform mass and area or do they include the equipment?